apps/react-18-tests-v8/src/App.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+import { ThemeProvider, DefaultButton, PartialTheme, getTheme } from '@fluentui/react';
+import { useBoolean } from '@fluentui/react-hooks';
+import { ContextualMenuExample } from './components';
+
+// This app is here as a simple sandbox to render v8 controls inside of an React 18 environment.
+
+export const App = () => {
+  const { palette } = getTheme();
+
+  const lightTheme: PartialTheme = {
+    semanticColors: {
+      bodyBackground: palette.white,
+      bodyText: palette.black,
+      menuBackground: palette.white,
+    },
+  };
+
+  const darkTheme: PartialTheme = {
+    semanticColors: {
+      bodyBackground: palette.black,
+      bodyText: palette.white,
+      menuBackground: palette.black,
+    },
+  };
+  const [isLight, { toggle: toggleIsLight }] = useBoolean(true);
+
+  const ToggleButton: React.FunctionComponent = () => {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore typing error caused by monorepo being on @types/react version 17 and this app being on react 18.
+    // Seems to be caused by certain v8 components (like Button) still being class components.
+    // Error goes away when monorepo is migrated to react 18 types OR when @fluentui/react increases peer dependency
+    // range to include react 18.
```

Comment:
1) `Button` components show this typing error is caused by the the app using `@types/react` version 17 since the monorepo is on 17 thus causing a typing conflict since `ReactFragment` no longer has `{}` in its declaration which is used by `ReactNode`. From what I tried, this issue was only resolved when `@types/react` for the monorepo was upgraded to version 18 or when `@fluentui/react`'s peer dependency range was expanded to include react 18 so decided to just add `ts-ignore` here since it's not really an issue with the component itself
```
'DefaultButton' cannot be used as a JSX component.
  Its instance type 'DefaultButton' is not a valid JSX element.
    Types of property 'refs' are incompatible.
      Type '{ [key: string]: import("/home/tristanwatanabe/Documents/FluentUI/v8-buildwork/node_modules/@types/react/index").ReactInstance; }' is not assignable to type '{ [key: string]: React.ReactInstance; }'.
        'string' index signatures are incompatible.
          Type 'import("/home/tristanwatanabe/Documents/FluentUI/v8-buildwork/node_modules/@types/react/index").ReactInstance' is not assignable to type 'React.ReactInstance'.
            Type 'Component<any, {}, any>' is not assignable to type 'ReactInstance'.
              Type 'import("/home/tristanwatanabe/Documents/FluentUI/v8-buildwork/node_modules/@types/react/index").Component<any, {}, any>' is not assignable to type 'React.Component<any, {}, any>'.
                The types returned by 'render()' are incompatible between these types.
                  Type 'import("/home/tristanwatanabe/Documents/FluentUI/v8-buildwork/node_modules/@types/react/index").ReactNode' is not assignable to type 'React.ReactNode'.
                    Type '{}' is not assignable to type 'ReactNode'.ts(2786)```